File: test/testmovieGenre.js
// module = require("./movieGenre")
// let movie = Movie();
// movie = Movie()
// darkWorld = new Movie()
// thor = new Movie()
// action = new Genre()
// action.name = "Action"
// darkWorld.title = "Dark World"
// thor.title = "Thor"
// thor.genres = action.name
// darkWorld.genres = action.name
// darkWorld.addRelatedMovie(thor);
// thor.addRelatedMovie(darkWorld);

// console.log(thor.related[0].title)
// console.log(thor.genres)

//const Movie = require('./movieGenre').Movie;
//const Genre = require('./movieGenre').Genre;
//const assert = require('assert');

//module = require('./movieGenre.js')


let movie = require("../js/movieGenre");
let Movie = movie.Movie;
let genre = require("../js/movieGenre");
let Genre = genre.Genre;


aquaman = new Movie();
aquaman.uuid = '001';
aquaman.title = 'Aquaman';
aquaman.year = 2018;

bumblebee = new Movie();
bumblebee.uuid = '002';
bumblebee. title = 'Bumblebee';
bumblebee.year = 2018;


console.log('addRelatedMovie Test Cases');
function test_addRelated(a, b){
   
   if (a.addRelatedMovie(b) === true){
   
       console.log('TEST PASSED');
   } else {
       console.log('TEST FAILED');
   }
}
test_addRelated(aquaman, bumblebee); 

File: test/testMovieGenre.py

# # darkWorld = Movie()
# # darkWorld.title = "Dark World"
# # print(thor.addRelatedMovie(darkWorld))
# # print(thor.related[0].title)


from movieGenre import *

thor = new Movie()
thor.uuid = '003'
thor.title = 'Thor: Ragnarok night'
thor.year = 2017


logan = new Movie()
logan.uuid = '004'
logan. title = 'Logan'
logan.year = 2017


wonderwoman = new Movie()
wonderwoman.uuid = '003'
wonderwoman.title = 'Wonder Woman'
wonderwoman.year = 2017


hero = new Genre()
hero.name = 'Hero'

action = new Genre()
action.name = 'Action'

fantasy = new Genre()
fantasy.name = 'Fantasy'


print('Add Related Movie')
def test_addRelated(a, b):
    if (a.addRelatedMovie(b) == True):
        print('true')
    else:
        print('false')
test_addRelated(thor, logan)



print('\n setGenre ')
def test_setGenre(m, g):
    if (m.setGenre(g) == True):
        m.genres = g.name
        print('True')
    else:
        print('TEST FAILED')
test_setGenre(thor, action)
test_setGenre(logan, hero)


# addMovie Test Cases
print('\n addMovie ')
def test_addMovie(g, m):
    if (g.addMovie(m) == True):
        print('True')
    else:
        print('False')
test_addMovie(fantasy, wonderwoman)
# test_addMovie(hero, superman); 


# display movies
# print('\n', thor)
# print('\n', avengers)
# print('\n', spiderman)

# display genres
# print('\n', fantasy)
# print('\n', hero)
# print('\n', action)
        